src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,184 @@
+    for key in keys:
+        restricted_units_dict[key] = list(set(restricted_units_dict[key]))
+    return restricted_units_dict
+
+
+restricted_units = build_restricted_conversions(
+    restricted_conversions, restricted_unit_expansion
+)
+
+
+class Quantity:
+
+    """This class instantiates physical quantities using their real values and
+    units.
+
+    All the instances of this class are converted to base SI units
+    system. Any conversion between "Hz", "hertz", "rad/s", "radian/s",
+    "rpm", "rps", "cps" is restricted.
+    """
+
+    def __init__(self, real_value, units_string):
+        self.value = real_value
+        self.unit = units_string
+        self._quantity = quantity(self.value, self.unit)
+        self._base_si_quantity = self._quantity.to_base_units()
+        self._restricted_conversions = restricted_units
+
+    def __float__(self):
+        return Quantity(self.value, self.unit)
+
+    def __str__(self):
+        return f'({self.value}, "{self.unit}")'
+
+    def __repr__(self):
+        return f'(Quantity ({self.value}, "{self.unit}"))'
+
+    def to(self, to_unit):
+
+        """This method checks the compatibility between current instance unit
+        and user provided unit, if both of them are compatible, then only it
+        performs required conversion otherwise raises a Value Error."""
+
+        if (
+            self.unit in self._restricted_conversions.keys()
+            and to_unit in self._restricted_conversions[self.unit]
+        ):
+            raise ValueError(
+                f"Conversion between '{self.unit}' and '{to_unit}' is restricted."
+            )
+
+        user_unit = Unit(to_unit)
+
+        if not self._quantity.is_compatible_with(user_unit):
+            raise ValueError("Units are not compatible")
+
+        converted = self._quantity.to(to_unit)
+
+        return Quantity(converted.magnitude, to_unit)
+
+    def __mul__(self, other):
+
+        if isinstance(other, Quantity):
+            temp = self._base_si_quantity * other._quantity
+        elif isinstance(other, int) or isinstance(other, float):
+            temp = quantity(self._base_si_quantity * other, self.unit)
+        return Quantity(temp.magnitude, temp.units)
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+
+        if isinstance(other, Quantity):
+            temp = self._base_si_quantity / other._quantity
+        elif isinstance(other, int) or isinstance(other, float):
+            temp = quantity(self._base_si_quantity / other, self.unit)
+        return Quantity(temp.magnitude, temp.units)
+
+    def __add__(self, other):
+        if self.unit in self._restricted_conversions.keys():
+            raise ValueError("This arithmetic operation is restricted")
+
+        if isinstance(other, Quantity):
+            temp = self._base_si_quantity + other._quantity
+        elif isinstance(other, int) or isinstance(other, float):
```

Comment:
Do we not need to check that the object is dimensionless before allowing the addition operation with float or int?

Reply:
Yes, pint supports method to check whether the object is dimensionless or not and returns boolean as follows -

![image](https://user-images.githubusercontent.com/106588300/186350072-467b3101-a9e6-4ff2-a85b-acfa187c61ee.png)

Reply:
Yes, all tests are still passing by implementing following change in ```__add__, __radd__, __sub__, __rsub__``` methods.

![image](https://user-images.githubusercontent.com/106588300/186362994-ecc65efe-0f9f-45f5-9806-222f58e7a731.png)

![image](https://user-images.githubusercontent.com/106588300/186364102-1b4e93dc-c4aa-48c4-ab13-27e7fbe8f471.png)